### ISSUE TITLE
987k non 2xx responses

### ISSUE THREAD

**Issue by Eilie** [iron/iron#534]
Hello,

Tried to benchmark Iron with wrk and got following output:
```
 ~ wrk --latency -t12 -c100 -d10s http://localhost:3000/name/foo
Running 10s test @ http://localhost:3000/name/foo
  12 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   355.61us  798.10us  46.90ms   98.89%
    Req/Sec    24.58k     4.16k   47.21k    86.14%
  Latency Distribution
     50%  281.00us
     75%  414.00us
     90%  571.00us
     99%    1.20ms
  987825 requests in 10.09s, 77.25MB read
  Non-2xx or 3xx responses: 987825
Requests/sec:  97895.62
Transfer/sec:      7.66MB
```
What does following line mean?
```
Non-2xx or 3xx responses: 987825
```

**Comment by untitaker** [comment]
With which iron app did you test that?

**Comment by Eilie** [comment]
@untitaker https://github.com/iron/iron/blob/master/examples/hello.rs

**Comment by untitaker** [comment]
I don't get those stats on my local machine using the same command as you did. Can you access the URL you used for wrk from your browser? This may be one of those issues where Iron listens to the  ipv6 localhost (`[::1]`), but wrk sends their request to ipv4 localhost (`127.0.0.1`). Replace `localhost` with an IP address (`127.0.0.1`), both in your app and the wrk command to work around this.

**Comment by Eilie** [comment]
@untitaker Alright, I was using old binary(building with `--release` manually) so hello.rs works fine indeed. I'm sorry. The code that triggers such response is following:
```
extern crate iron;
#[macro_use]
extern crate router;

use iron::prelude::*;
use iron::status;
use router::Router;

fn hello_controller(req: &mut Request) -> IronResult<Response> {
    let ref query = req.extensions.get::<Router>().unwrap().find("name").unwrap();
    Ok(Response::with((status::Ok, *query)))
}

fn main() {
    let router = router!(
        hello: get "/hello/:name" => hello_controller
    );

    Iron::new(router).http("127.0.0.1:3000").unwrap();
}
```

Building it with `--release` and wrk command is `wrk --latency -t12 -c100 -d10s http://127.0.0.1:3000/name/foo`

**Comment by sfackler** [comment]
http://127.0.0.1:3000/name/foo is going to hit a 404, which is not a 2xx or 3xx response code.

**Comment by untitaker** [comment]
Steven: I thought that too, but since the helloworld example is not using a
router, it accepts any URL path.

**Comment by untitaker** [comment]
Ah, but you are right for the example @Eilie posted.

**Comment by Eilie** [comment]
@sfackler @untitaker Ohhh.. Feeling retarded, sorry about that. That explains everyhting, thank you guys a lot!

**Comment by untitaker** [comment]
Great to hear, and no problem :)

On April 13, 2017 8:11:15 PM GMT+02:00, Eilie <notifications@github.com> wrote:
>Closed #534.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/iron/iron/issues/534#event-1042363390

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
